src/test/java/io/odh/test/e2e/upgrade/OlmUpgradeST.java

```diff
@@ -113,8 +138,38 @@ void testUpgradeOlm() throws IOException, InterruptedException {
         Instant operatorLogCheckTimestamp = Instant.now();
 
         // Verify that NTB pods are stable
-        PodUtils.waitForPodsReady(ntbNamespace, lblSelector, 1, true, () -> { });
+        PodUtils.waitForPodsReady(ntbNamespace, lblSelectorPre, 1, true, () -> { });
         // Check logs in operator pod
         UpgradeUtils.deploymentLogIsErrorEmpty(olmInstall.getNamespace(), olmInstall.getDeploymentName(), operatorLogCheckTimestamp);
+
+        // RHOAIENG-10827: creating new notebooks after upgrade restarted existing notebooks
+        Allure.step("Deploy another notebook after upgrade");
+        deployNotebook(ntbNamespace, ntbNamePost);
+
+        LabelSelector lblSelectorPost = new LabelSelectorBuilder()
+                .withMatchLabels(Map.of("app", ntbNamePost))
+                .build();
+        PodUtils.waitForPodsReady(ntbNamespace, lblSelectorPost, 1, true, () -> { });
+
+        // Verify that old notebook pod is running
+        PodUtils.waitForPodsReady(ntbNamespace, lblSelectorPre, 1, true, () -> { });
+        // and
+        List<Pod> ntbPodsPreAfter = KubeResourceManager.getKubeClient().getClient().pods()
+                .inNamespace(ntbNamespace).withLabelSelector(lblSelectorPre).list().getItems();
+        Assertions.assertEquals(1, ntbPodsPreAfter.size());
+        // has not been restarted
+        for (ContainerStatus containerStatus : ntbPodsPreAfter.get(0).getStatus().getContainerStatuses()) {
+            Assertions.assertEquals(0, containerStatus.getRestartCount(), containerStatus.toString());
+        }
+        // nor recreated
+        Assertions.assertEquals(ntbPodsPre.get(0).getMetadata().getUid(), ntbPodsPreAfter.get(0).getMetadata().getUid());
+        // nor modified in any way
+        Assertions.assertEquals(ntbPodsPre.get(0).getMetadata().getResourceVersion(), ntbPodsPreAfter.get(0).getMetadata().getResourceVersion());
+
+        // Verify that old notebook resource spec has not been modified
+        Notebook ntbResourcePreAfter = KubeResourceManager.getKubeClient().getClient().resources(Notebook.class).inNamespace(ntbNamespace).withName(ntbNamePre).get();
+        Assertions.assertEquals(ntbResourcePre.getMetadata().getUid(), ntbResourcePreAfter.getMetadata().getUid());
+        Assertions.assertEquals(ntbResourcePre.getSpec(), ntbResourcePreAfter.getSpec());
+        Assertions.assertEquals(ntbResourcePre.getMetadata().getResourceVersion(), ntbResourcePreAfter.getMetadata().getResourceVersion());
```

Comment:
this may need adjusting in the future, I believe the assertion is not always correct because the activity probes for culler update nb status and can therefore change resource version; I did not see this assertion fail in jenkins so I'm keeping it there for now

Reply:
* https://github.com/skodjob/odh-e2e/pull/166